library/ui/src/main/java/com/google/android/exoplayer2/ui/PlayerControlView.java

```diff
@@ -835,13 +839,33 @@ private void updateProgress() {
         if (playbackSpeed <= 0.1f) {
           delayMs = 1000;
         } else if (playbackSpeed <= 5f) {
-          long mediaTimeUpdatePeriodMs = 1000 / Math.max(1, Math.round(1 / playbackSpeed));
+
+          int timeBarWidth = timeBar.getTimeBarWidth();
+          // Calculate how many updates needs to be done with DEFAULT_UPDATE_DP
+          // to fill up the time bar
+          int numberOfUpdates = timeBarWidth / DEFAULT_UPDATE_DP;
+
+          // Calculate the designated update interval, taking duration into consideration as well
+          long mediaTimeUpdatePeriodMs = duration / numberOfUpdates;
+
+          // Limit the designated update interval, to avoid too frequent / infrequent updates
```

Comment:
I think that should happen at the very end after taking playback speed into account. Otherwise the playback speed may still scale the delay to very large or small values.

Reply:
Be aware though, that the duration can be 0 (e.g. for live streams with unset duration).

Reply:
Same is probably true for the timeBarWidth? Could be 0 if the view is embedded in a strange way where the the timebar isn't really visible anymore.